src/datatrove/tools/word_tokenizers.py

```diff
@@ -0,0 +1,197 @@
+    Languages.hebrew: SpaCyTokenizer("he"),
+    Languages.latvian: SpaCyTokenizer("lv"),
+    Languages.icelandic: SpaCyTokenizer("is"),
+    Languages.armenian: SpaCyTokenizer("hy"),
+    Languages.basque: SpaCyTokenizer("eu"),
+    Languages.thai: ThaiTokenizer(),
+    Languages.georgian: GeorgianTokenizer(),
+    Languages.tagalog: SpaCyTokenizer("tl"),
+    Languages.albanian: SpaCyTokenizer("sq"),
+    Languages.macedonian: SpaCyTokenizer("mk"),
+    Languages.azerbaijani: SpaCyTokenizer("az"),
+    Languages.amharic: SpaCyTokenizer("am"),
+    Languages.bengali: SpaCyTokenizer("bn"),
+    Languages.malay: SpaCyTokenizer("ms"),
+    Languages.urdu: SpaCyTokenizer("ur"),
+    Languages.nepali: SpaCyTokenizer("ne"),
+    Languages.kazakh: StanzaWordTokenizer("kk"),
+    Languages.gujarati: SpaCyTokenizer("gu"),
+    Languages.kannada: SpaCyTokenizer("kn"),
+    Languages.welsh: StanzaWordTokenizer("cy"),
+    Languages.norwegian_nynorsk: NLTKTokenizer("norwegian"),
+    Languages.sinhala: SpaCyTokenizer("si"),
+    Languages.tatar: SpaCyTokenizer("tt"),
+    Languages.afrikaans: SpaCyTokenizer("af"),
+    Languages.kirghiz: SpaCyTokenizer("ky"),
+    Languages.irish: SpaCyTokenizer("ga"),
+    Languages.luxembourgish: SpaCyTokenizer("lb"),
+    Languages.maltese: StanzaWordTokenizer("mt"),
+    Languages.sanskrit: SpaCyTokenizer("sa"),
+    Languages.yoruba: SpaCyTokenizer("yo"),
+    Languages.pashto: PashtoTokenizer(),
+    Languages.serbocroatian: SpaCyTokenizer("sr"),
+    Languages.oriya: IndicNLPTokenizer("or"),
+    Languages.punjabi: IndicNLPTokenizer("sa"),
+    Languages.assamese: IndicNLPTokenizer("as"),
+    Languages.war: IndicNLPTokenizer("war"),
+    Languages.sindhi: IndicNLPTokenizer("sd"),
+    Languages.bosnian: SpaCyTokenizer("hr"),  # Proxy
+    Languages.belarusian: SpaCyTokenizer("uk"),  # Proxy
+    Languages.galician: NLTKTokenizer("portuguese"),  # Proxy
+    Languages.esperanto: NLTKTokenizer("english"),  # Proxy
+    Languages.occitan: NLTKTokenizer("italian"),  # Proxy
+    Languages.cebuano: NLTKTokenizer("english"),  # Proxy
+    Languages.swahili: NLTKTokenizer("english"),  # Proxy
+    Languages.javanese: NLTKTokenizer("english"),  # Proxy
+    Languages.uzbek: NLTKTokenizer("turkish"),  # Proxy, alternative ru
+    Languages.tajik: SpaCyTokenizer("ru"),  # Proxy
+    Languages.kurdish: NLTKTokenizer("english"),  # Proxy, multiple scripts!
+    Languages.sorani: SpaCyTokenizer("fa"),  # Proxy
+    Languages.south_azerbaijani: SpaCyTokenizer("fa"),  # Proxy
+    Languages.bashkir: SpaCyTokenizer("tt"),  # Proxy
+    Languages.western_frisian: NLTKTokenizer("dutch"),  # Proxy
+    Languages.breton: StanzaWordTokenizer("cy"),  # Proxy
+    Languages.malagasy: NLTKTokenizer("english"),  # Proxy
+    Languages.yiddish: SpaCyTokenizer("he"),  # Proxy
+    Languages.somali: NLTKTokenizer("english"),  # Proxy
+    Languages.turkmen: NLTKTokenizer("turkish"),  # Proxy
+}
+
+
+def get_word_tokenizer(language: str):
```

Comment:
This code is currently `dead code`. Can we add a test that guarantees that the behavior for English does not change and then replace:

```
        from nltk.tokenize import word_tokenize

        words = word_tokenize(doc.text)
   ```
  with automatic language specific tokenisation?

Reply:
Currently, the tests in [`tests/pipeline/test_filters.py`](https://github.com/huggingface/datatrove/blob/afadc8f20f7666b86743a9554d231c0177b2f39f/tests/pipeline/test_filters.py#L37) don't include language metadata. Including automatic language specific tokenization in the filters makes the tests fail. Should we also modify the tests to support the change?